dashboard/app/controllers/codeprojects_preview_controller.rb

```diff
@@ -1,6 +1,30 @@
 class CodeprojectsPreviewController < ApplicationController
+  include AllowedHostnameHelper
   # Public preview page, static content for now.
   def show
+    code_studio_url = CDO.dashboard_site_host
+    allowed_connect_src = ALLOWED_HOSTNAME_SUFFIXES.join(" ")
+
+    if rack_env?(:development)
+      # dashboard_site_host is set to use port 3000 in development, but we want to also allow port 9000.
+      port_9000_url = code_studio_url.split(":").first + ":9000"
+      code_studio_url += " #{port_9000_url}"
+    end
+
+    policies = [
```

Comment:
Kind of a specific question / can probably figure it out down the line, but do you know if we'll need FontAwesome? Asking b/c I think it would be loaded via dsco.code.org, and I see there's no font-src here.

Reply:
Good question, probably not? I don't think students would use font awesome in their projects, nor would we teach it.

Reply:
Agh sorry I think I got my brain twisted between our code v. student code. SGTM!